TUSKit/Classes/TUSExecutor.swift

```diff
@@ -180,14 +180,10 @@ class TUSExecutor: NSObject, URLSessionDelegate {
                             self.handleUploadSuccess(upload: upload, completion: completion)
                         }
                     }
-                case 400 ..< 500:
+                case 300 ..< 600:
                     // reuqest error
                     TUSClient.shared.logger.log(forLevel: .Error, withMessage: String(format: "Received request failure status code %u", httpResponse.statusCode))
-                    self.markAsFailed(upload: upload, completion: completion, error: nil)
-                case 500 ..< 600:
-                    // server
-                    TUSClient.shared.logger.log(forLevel: .Error, withMessage: String(format: "Received server failure status code %u", httpResponse.statusCode))
-                    self.markAsFailed(upload: upload, completion: completion, error: nil)
```

Comment:
i did this change as the errors were handled exactly the same (DRY)

Reply:
3xx doesn't directly indicate an http error, but according to the TUS protocol it is not used a "success" HTTP code, thus should be treated as error